parser.go

```diff
@@ -12,16 +12,26 @@ type Parser struct {
 	// If populated, only these methods will be considered valid.
 	validMethods []string
 
-	// Use JSON Number format in JSON decoder.
-	useJSONNumber bool
-
 	// Skip claims validation during token parsing.
 	skipClaimsValidation bool
 
 	validator *Validator
 
-	decodeStrict bool
+	decoding
+}
+
+type decoding struct {
```

Comment:
its not private, but still we probably could use some comments here, especially for this outer struct